tests/Unit/Adapter/LegacyDeliveryOptionsAdapterTest.php

```diff
@@ -0,0 +1,82 @@
+
+namespace MyParcelNL\WooCommerce\Adapter;
+
+use MyParcelNL\Pdk\Facade\Pdk;
+use MyParcelNL\Pdk\Shipment\Model\DeliveryOptions;
+use MyParcelNL\WooCommerce\Tests\Uses\UsesMockWcPdkInstance;
+use function MyParcelNL\Pdk\Tests\factory;
+use function MyParcelNL\Pdk\Tests\usesShared;
+use function Spatie\Snapshots\assertMatchesSnapshot;
+
+usesShared(new UsesMockWcPdkInstance());
+
+dataset('deliveryOptions', function () {
+    return [
+        'with date'        => function () {
+            return factory(DeliveryOptions::class)
+                ->with([
+                    'deliveryType' => 'standard',
+                    'packageType'  => 'package',
+                    'carrier'      => 'postnl',
+                    'date'         => '2037-12-31',
+                ])
+                ->make();
+        },
+        'carrier'          => function () {
+            return factory(DeliveryOptions::class)
+                ->with([
+                    'deliveryType' => 'standard',
+                    'packageType'  => 'mailbox',
+                    'carrier'      => 'dhlforyou',
+                ])
+                ->make();
+        },
+        'shipment options' => function () {
+            return factory(DeliveryOptions::class)
+                ->with([
+                    'deliveryType'    => 'standard',
+                    'packageType'     => 'package',
+                    'carrier'         => 'postnl',
+                    'shipmentOptions' => [
+                        'ageCheck'      => true,
+                        'signature'     => true,
+                        'onlyRecipient' => false,
+                    ],
+                ])
+                ->make();
+        },
+        'pickup location'  => function () {
+            return factory(DeliveryOptions::class)
+                ->with([
+                    'deliveryType'   => 'pickup',
+                    'packageType'    => 'package',
+                    'carrier'        => 'dpd',
+                    'pickupLocation' => [
+                        'locationCode'    => 'DPD-12',
+                        'locationName'    => 'DPD Pakketshop',
+                        'retailNetworkId' => '123',
+                        'street'          => 'Deepeedee',
+                        'number'          => '12',
+                        'postalCode'      => '1212DP',
+                        'city'            => 'Hoofddorp',
+                        'country'         => 'NL',
+                    ],
+                ])
+                ->make();
+        },
+    ];
+});
+
+it('creates legacy options', function (DeliveryOptions $options) {
+    /** @var LegacyDeliveryOptionsAdapter $adapter */
+    $adapter = Pdk::get(LegacyDeliveryOptionsAdapter::class);
+
+    /**
+     * In the snapshots, properties in pickupLocation and shipmentOptions must be snake_case (part of the legacy)
+     */
+    assertMatchesSnapshot($adapter->fromDeliveryOptions($options));
```

Comment:
moet dit een snapshot test zijn? Ik weet dat het makkelijk is om te doen en maak mezelf er ook vaak genoeg schuldig aan, maar dit kan toch ook gewoon een "_normale_" test zijn?

Reply:
Liever snapshots, omdat je anders toch ergens soort van hard-coded die snake case moet intypen.